### ISSUE TITLE
Clippy complains when const_format is used

### ISSUE THREAD

**Issue by alopatindev** [rodrimati1992/const_format_crates#7]
```rust
const A: &str = "A";
const B: &str = const_format::concatcp!(A, "B");

fn main() {
    println!("{}", B);
}
```

```
warning: unsequenced read of a variable
 --> src/main.rs:2:17
  |
2 | const B: &str = const_format::concatcp!(A, "B");
  |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(clippy::eval_order_dependence)]` on by default
note: whether read occurs before this write depends on evaluation order
 --> src/main.rs:2:17
  |
2 | const B: &str = const_format::concatcp!(A, "B");
  |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#eval_order_dependence
  = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: unsequenced read of a variable
 --> src/main.rs:2:17
  |
2 | const B: &str = const_format::concatcp!(A, "B");
  |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
note: whether read occurs before this write depends on evaluation order
 --> src/main.rs:2:17
  |
2 | const B: &str = const_format::concatcp!(A, "B");
  |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#eval_order_dependence
  = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: transmute from a reference to a reference
 --> src/main.rs:2:17
  |
2 | const B: &str = const_format::concatcp!(A, "B");
  |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(clippy::transmute_ptr_to_ptr)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#transmute_ptr_to_ptr
  = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: 3 warnings emitted
```

**Comment by alopatindev** [comment]
Adding `#[allow(clippy::eval_order_dependence, clippy::transmute_ptr_to_ptr)]` above each affected line fixes that, but that's quite noisy solution

**Comment by rodrimati1992** [comment]
Alright, I fixed these warnings in #8 , doing a patch release in a few hours.

**Comment by rodrimati1992** [comment]
If you're curious, none of the warnings indicated anything wrong about correctness, the ones I fixed (instead of silencing them) were about improving the code.

**Comment by rodrimati1992** [comment]
Released the [0.2.7](https://crates.io/crates/const_format/0.2.7) version with the fix.
